### ISSUE TITLE
Make compiler output independent on execution environment

### ISSUE THREAD

**Issue by Schaeff** [Zokrates/ZoKrates#423]
We output solvers in a way that exposes implementation details of the execution environment: for example `Rust::Bits`.

Instead, we should have something like `Solver::Bits`, `Solver::Wasm(wasmcode)` and the EE is has its own implementation of these things.

Low priority, but it's confusing to have Rust in there where in theory the compiler output is not Rust specific at all.

**Comment by fulldecent** [comment]
Possibly related. But I consider the .code output to be a debug build artifact (the file that contains the `Rust::Bits` per above). Most end-users will not use that file. 

Should generating the .code file be disabled by default?

**Comment by Schaeff** [comment]
I agree we need to make it clearer this is only debug, however this issue is more than that as the `out` which is not debug still has `Rust::Bits` in it where it should be `Bits` and the interpreter has its implementation.
